cluster/kubernetes/resourcekinds.go

```diff
@@ -176,23 +201,30 @@ func (dk *daemonSetKind) getPodControllers(c *Cluster, namespace string) ([]podC
 func makeDaemonSetPodController(daemonSet *apiapps.DaemonSet) podController {
 	var status string
 	objectMeta, daemonSetStatus := daemonSet.ObjectMeta, daemonSet.Status
+
+	status = StatusUpdating
+	pods := cluster.Pods{
+		Desired:   daemonSetStatus.DesiredNumberScheduled,
+		Updated:   daemonSetStatus.UpdatedNumberScheduled,
+		Ready:     daemonSetStatus.NumberReady,
+		Available: daemonSetStatus.NumberAvailable,
+		Outdated:  daemonSetStatus.CurrentNumberScheduled - daemonSetStatus.UpdatedNumberScheduled,
+	}
+
 	if daemonSetStatus.ObservedGeneration >= objectMeta.Generation {
 		// the definition has been updated; now let's see about the replicas
-		updated, wanted := daemonSetStatus.UpdatedNumberScheduled, daemonSetStatus.DesiredNumberScheduled
-		if updated == wanted {
+		status = StatusUpdating
+		if pods.Ready == pods.Desired && pods.Available == pods.Desired && pods.Outdated == 0 {
 			status = StatusReady
-		} else {
-			status = fmt.Sprintf("%d out of %d updated", updated, wanted)
 		}
-	} else {
-		status = StatusUpdating
 	}
 
 	return podController{
 		apiVersion:  "apps/v1",
 		kind:        "DaemonSet",
 		name:        daemonSet.ObjectMeta.Name,
 		status:      status,
+		pods:        pods,
```

Comment:
DaemonSets have conditions in the same way as deployments, how come we're not looking at those?